File: benchmark/2026-09-17-7677581(latest)/semantic-compression/probes/cpp/F15.P3.cpp
#include <cstdio>
#include <string>
#include <vector>

std::string first_tag() {
// BEGIN PROBE F15.P3
    struct Named { virtual std::string tag() const = 0; };
    struct A : Named { std::string tag() const override { return "a"; } };
    struct B : Named { std::string tag() const override { return "b"; } };
    A a;
    B b;
    std::vector<Named*> items{&a, &b};
    return items[0]->tag();
// END PROBE F15.P3
}

int main() {
    std::printf("%s\n", first_tag().c_str());
}

File: benchmark/2026-09-17-7677581(latest)/semantic-compression/probes/kotlin/F15.P3.kt
// BEGIN PROBE F15.P3
interface Named {
    fun tag(): String
}

class A : Named {
    override fun tag(): String = "a"
}

class B : Named {
    override fun tag(): String = "b"
}

fun firstTag(): String {
    val items: List<Named> = listOf(A(), B())
    return items[0].tag()
}
// END PROBE F15.P3

fun main() {
    println(firstTag())
}
